libs/perun/components/src/lib/settings-mailing-lists/mailing-lists.component.ts

```diff
@@ -0,0 +1,246 @@
+  AttributesManagerService,
+  InputSetMemberResourceAttribute,
+  MembersManagerService,
+  ResourcesManagerService,
+  RichResource,
+  User,
+  UsersManagerService,
+  Vo,
+} from '@perun-web-apps/perun/openapi';
+import { StoreService, NotificatorService } from '@perun-web-apps/perun/services';
+import { compareFnName } from '@perun-web-apps/perun/utils';
+import { TranslateService } from '@ngx-translate/core';
+
+@Component({
+  selector: 'perun-web-apps-mailing-lists',
+  templateUrl: './mailing-lists.component.html',
+  styleUrls: ['./mailing-lists.component.scss'],
+})
+export class MailingListsComponent implements OnInit, OnDestroy {
+  @Input() user: User;
+  @Input() isService: boolean;
+  vos: Vo[] = [];
+  resources: RichResource[] = [];
+  optOuts: InputSetMemberResourceAttribute[] = [];
+  optOutAttribute: Attribute;
+  index: number;
+  filteredVos: Vo[] = [];
+  loading = true;
+  selectedVo: string = null;
+  selectedResource: string = null;
+  changeOptOut: string;
+
+  constructor(
+    private store: StoreService,
+    private usersManagerService: UsersManagerService,
+    private membersService: MembersManagerService,
+    private resourcesManagerService: ResourcesManagerService,
+    private attributesManagerService: AttributesManagerService,
+    private route: ActivatedRoute,
+    private router: Router,
+    private notificator: NotificatorService,
+    private translate: TranslateService
+  ) {}
+
+  ngOnDestroy(): void {
+    void this.router.navigate([], {
+      relativeTo: this.route,
+      queryParams: { vo: null, resource: null },
+      replaceUrl: true,
+    });
+  }
+
+  ngOnInit(): void {
+    this.route.queryParams
+      .subscribe((params) => {
+        this.selectedVo = String(params['vo']);
+        this.selectedResource = String(params['resource']);
+        this.changeOptOut = String(params['action']);
+
+        this.usersManagerService.getVosWhereUserIsMember(this.user.id).subscribe((vos) => {
+          this.vos = vos.sort(compareFnName);
+          this.filteredVos = vos;
+          if (this.selectedResource !== undefined) {
+            const vo = this.vos.find((obj) => obj.shortName === this.selectedVo);
+            if (vo) {
+              this.getMailingLists(vo);
+            }
+          } else if (this.selectedVo !== undefined) {
+            const vo = this.vos.find((obj) => obj.shortName === this.selectedVo);
+            if (vo) {
+              this.getMailingLists(vo);
+              this.changeSelectedVo(vo);
+            }
+          }
+        });
+      })
+      .unsubscribe();
+  }
+
+  changeSelectedResource(resource: RichResource): void {
+    if (this.selectedResource !== resource.name) {
+      this.getOptOutAttribute(resource);
+    }
+    if (this.changeOptOut) {
+      if (this.changeOptOut === 'subscribe') {
+        this.subscribe();
+      } else if (this.changeOptOut === 'unsubscribe') {
+        this.unsubscribe();
+      }
+      this.changeOptOut = null;
+    }
```

Comment:
I would move this block of code to the if the block below (`if (!this.isService){...}`) because this way it would be possible to subscribe/unsuscribe the mailing list via the URL, but I think we don't want it in the admin GUI at this time.